engine/src/runtime/helpers.rs

```diff
@@ -53,20 +59,20 @@ pub fn get_memory_from_caller<T>(caller: &mut Caller<'_, T>) -> Result<Memory, (
 pub fn read_bytes<T>(
     caller: &Caller<'_, T>,
     memory: Memory,
-    ptr: u32,
-    len: u32,
+    ptr: usize,
```

Comment:
Honestly, I am not sure how this will play out if we have a mix of 32- and 64-bit WASM payloads. 🤔 This usize will be based on the host, not the guest, and I am not sure how those two things will interact.